### PR TITLE
[4.0] Convert plugin system stats to prepared statements

### PR DESCRIPTION
**Pull request by HLeithner**
# Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the plugin in all ways you can think of.

This Plugin sent statistics to Joomla.org, so using, prepared statements changes in the parameter save function. So saving the plugin settings should be one part.

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/system/stats/stats.php [hunk 1]
```diff
@@ -252,10 +252,10 @@ public function onAjaxSendStats()
 		// User has not selected the mode. Show message.
 		if ((int) $this->params->get('mode') !== static::MODE_ALLOW_ALWAYS)
 		{
-			$data = array(
+			$data = [
 				'sent' => 0,
 				'html' => $this->getRenderer('message')->render($this->getLayoutData())
```
Comment (SharkyKZ) [inline plugins/system/stats/stats.php]:
```suggestion
				'html' => $this->getRenderer('message')->render($this->getLayoutData()),
```

File: plugins/system/stats/stats.php [hunk 2]
```diff
@@ -312,11 +312,11 @@ public function debug($layoutId, $data = array())
 	 */
 	protected function getLayoutData()
 	{
-		return array(
+		return [
 			'plugin'       => $this,
 			'pluginParams' => $this->params,
 			'statsData'    => $this->getStatsData()
```
Comment (SharkyKZ) [inline plugins/system/stats/stats.php]:
```suggestion
			'statsData'    => $this->getStatsData(),
```

File: plugins/system/stats/stats.php [hunk 3]
```diff
@@ -363,14 +363,14 @@ protected function getRenderer($layoutId = 'default')
...
 	private function getStatsData()
 	{
-		return array(
+		return [
 			'unique_id'   => $this->getUniqueId(),
 			'php_version' => PHP_VERSION,
 			'db_type'     => $this->db->name,
 			'db_version'  => $this->db->getVersion(),
 			'cms_version' => JVERSION,
 			'server_os'   => php_uname('s') . ' ' . php_uname('r')
```
Comment (SharkyKZ) [inline plugins/system/stats/stats.php]:
```suggestion
			'server_os'   => php_uname('s') . ' ' . php_uname('r'),
```

File: plugins/system/stats/stats.php [hunk 4]
```diff
@@ -588,10 +592,10 @@ private function clearCacheGroups(array $clearGroups)
 		{
 			try
 			{
-				$options = array(
+				$options = [
 					'defaultgroup' => $group,
 					'cachebase'    => $this->app->get('cache_path', JPATH_CACHE)
```
Comment (SharkyKZ) [inline plugins/system/stats/stats.php]:
```suggestion
					'cachebase'    => $this->app->get('cache_path', JPATH_CACHE),
```

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 0716853ff9a414d109f4c2997d01795efa969dd1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25541">issues.joomla.org/tracker/joomla-cms/25541</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 0716853ff9a414d109f4c2997d01795efa969dd1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25541">issues.joomla.org/tracker/joomla-cms/25541</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25541">issues.joomla.org/tracker/joomla-cms/25541</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**